Update information on autoconfig for Dynatrace Operator (#6)
dynatrace-oss-contrib/micrometer-docs: src/docs/implementations/dynatrace.adoc

For setting up new integrations with Dynatrace, it is recommended to use the latest version of the https://www.dynatrace.com/support/help/dynatrace-api/environment-api/metric-v2/[Dynatrace Metrics API] (v2).
Dynatrace provides different ways of setting up integrations:

> === Using Dynatrace autoconfiguration (preferred) [[bookmark-autoconfiguration]]

Reviewer: ```suggestion
=== Using Dynatrace auto-configuration (preferred) [[bookmark-auto-configuration]]
```
I think that's the more common spelling.

> Dynatrace autoconfiguration is available for hosts that are monitored by a OneAgent or by the Dynatrace Kubernetes Operator.

Reviewer: ```suggestion
Dynatrace auto-configuration is available for hosts that are monitored by a OneAgent or by the Dynatrace Operator for Kubernetes.
```

If a Dynatrace OneAgent is installed on the host running Micrometer, metrics can be exported directly using the OneAgent without having to specify an endpoint URI or API token.

> If running in Kubernetes with the Dynatrace operator installed, the registry will pick up your Dynatrace cluster endpoint from the operator instead.

Reviewer: ```suggestion
If running in Kubernetes with the Dynatrace operator installed, the registry will pick up your Dynatrace cluster endpoint URI and API token from the operator instead.
```

This allows configuring the Dynatrace exporter by using the <<bookmark-available-properties, the available properties>>.

To use the Dynatrace metrics exporter for Micrometer in your Spring Boot project, it is enough to include the `runtimeOnly 'io.micrometer:micrometer-registry-dynatrace'` dependency.
In this default configuration, metrics will be exported to the local OneAgent or Kubernetes Operator-provided endpoint.

=== Using a custom endpoint

> If autoconfiguration is not available on the host, both the Dynatrace Metrics API v2 endpoint and an API token have to be specified.

Reviewer: ```suggestion
If auto-configuration is not available on the host, both the Dynatrace Metrics API v2 endpoint and an API token have to be specified.
```

When the API version is configured to `v2`, the registry will send data using the https://www.dynatrace.com/support/help/dynatrace-api/environment-api/metric-v2/[Metrics API v2].
In order to maintain backwards compatibility, when a `deviceId` is set (which is required for `v1` and not used in `v2`), `v1` is used as the default.
Otherwise, the version defaults to `v2`, and does not have to be set explicitly.
With no endpoint URI and token set, metrics will be exported to the local OneAgent endpoint or, if a running in Kubernetes with the Dynatrace operator installed, to the endpoint provided by the operator.
If no auto-configuration is desired, it is possible to specify endpoint and token explicitly, in order to export metrics to that specific endpoint.
Explicitly specifying these will overwrite auto-configuration.

*Minimal configuration with Dynatrace auto-configuration*

> In the minimal configuration <<bookmark-auto-configuration, shown above>> (no URI or API token), the v2 exporter will attempt to retrieve the endpoint provided by the Dynatrace Kubernetes operator.

Reviewer: ```suggestion
In the minimal configuration <<bookmark-auto-configuration, shown above>> (no URI or API token), the v2 registry will attempt to retrieve the endpoint provided by the Dynatrace Kubernetes operator.
```

When the API version is configured to `v2`, the registry will send data using the https://www.dynatrace.com/support/help/dynatrace-api/environment-api/metric-v2/[Metrics API v2].
In order to maintain backwards compatibility, when a `deviceId` is set (which is required for `v1` and not used in `v2`), `v1` is used as the default.
Otherwise, the version defaults to `v2`, and does not have to be set explicitly.

> With no endpoint URL and token set, metrics will be exported to the local OneAgent endpoint or, if a running in Kubernetes with the Dynatrace operator installed, to the endpoint provided by the operator.

Reviewer: ```suggestion
With no endpoint URI and token set, metrics will be exported to the local OneAgent endpoint or, if a running in Kubernetes with the Dynatrace operator installed, to the endpoint provided by the operator.
```

We say URI everywhere else.

> If neither OneAgent nor Operator configurations are present, it is possible to specify endpoint and token explicitly, in order to export metrics to that specific endpoint.

Reviewer: This gets the precedence wrong actually.
